tests/cxx/namespace_use.cc

```diff
@@ -0,0 +1,45 @@
+//===--- namespace_use.cc - test input file for iwyu ----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I .
+
+// Tests that using-directives i.e. `using namespace`, get appropriate
+// include suggestions.
+
+#include "tests/cxx/namespace_use-i1.h"
+
+// Purely using the namespace, not any symbols.
+// This could be unused, or being used transitively in namespace lookup.
+// The include suggestion will be the first header that declares the namespace.
+// IWYU: i1_ns is defined in ...* which isn't directly #included.
+using namespace i1_ns;
+
+// This namespace gets used for symbols within it.
+// Suggestions for includes will come via the symbol usage rather than here.
+using namespace i2_ns;
+
+int main(int, const char**) {
+  // IWYU: i2_ns::get_value is defined in ...* which isn't directly #included.
```

Comment:
Here I think you _do_ know which filename, though? Such an assertion is usually spelled like this: https://github.com/include-what-you-use/include-what-you-use/blob/9e77b739be7014de0f8d17f86669cf8331e86337/tests/cxx/mapping_regex.cc#L27